test/box/gh-5518-add-granularity-option.test.lua

```diff
@@ -0,0 +1,204 @@
+
+-- write data recover from latest snapshot
+env = require('test_run')
+test_run = env.new()
+
+test_run:cmd("setopt delimiter ';'")
+test_run:cmd('create server test with script=\z
+             "box/gh-5518-add-granularity-option.lua"');
+test_run:cmd("setopt delimiter ''");
+-- Start server test with granularity == 2 failed
+-- (must be greater than or equal to 4)
+test_run:cmd('start server test with args="2" with crash_expected=True')
+-- Start server test with granularity == 7 failed (must be exponent of 2)
+test_run:cmd('start server test with args="7" with crash_expected=True')
+
+test_run:cmd('start server test with args="4"')
+test_run:cmd('switch test')
+
+s = box.schema.space.create('test')
+_ = s:create_index('test')
+
+-- Granularity determines not only alignment of objects,
+-- but also size of the objects in the pool. Thus, the greater
+-- the granularity, the greater the memory loss per one memory allocation,
+-- but tuples with different sizes are allocated from the same mempool,
+-- and we do not lose memory on the slabs, when we have highly
+-- distributed tuple sizes. This is somewhat similar to a large alloc factor
+
+-- Try to insert/delete to space, in case when UB sanitizer on,
+-- we check correct memory aligment
+for i = 1, 1000 do s:insert{i, i + 1} end
+-- Get item_size from box.slab.stats
+-- Show it and check that item_size is a multiple of granularity
+box_slab_stats = box.slab.stats()
+print("box_slab_stats for granularity = 4")
```

Comment:
As a rule we don't use prints in test. Better use asserts or just type variable to get its value in the result file.
Also the same chunk repeats several times for different granularity. Why not factor out it in a separate function?